proposed/error-result.md

```diff
@@ -0,0 +1,234 @@
+# DRAFT: Error and Result Handling
+
+## 1. Overview
+
+This document describes standard interfaces for representing operation results in PHP applications.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## 2. Definitions
+
+- **Result**: The outcome of an operation that may succeed or fail.
+- **Success Result**: A result containing a succesful value.
+- **Failure Result**: A result containing error information.
+- **Error**: Structured information about why an operation failed.
+
+## 3. Interfaces
+
+### 3.1 `ErrorInterface`
+
+```php
+<?php
+
+namespace Psr\Error;
+
+/**
+ * Represents structured error information.
+ */
+interface ErrorInterface extends \Throwable
```

Comment:
This seems problematic to me. One of the stated goals of this PSR is to avoid the overhead of Throwable/Exception, so it doesn't make sense for the error to extend Throwable.

To make it easier to "throw an error" then I would suggest adding a method `getException(): Throwable` (or `getThrowable`) to allow for:

```php
throw $error->getException();
```

Reply:
My bad, let's fix it

We have 2 ways:

1:

```php
interface ErrorInterface extends \Throwable
{
    public function getCode(): int | string;
}
```

2:

```php
interface ErrorInterface extends \Throwable
{
   public function getErrorCode(): string;
}
```

Which one is more acceptable?